safehttp/plugins/xsrf/xsrf.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package xsrf
+
+import (
+	"fmt"
+	"github.com/google/go-safeweb/safehttp"
+	"log"
+	// TODO(@empijei, @kele, @mattiasgrenfeldt, @mihalimara22): decide whether we want to depend on this package or reimplement thefunctionality
+	"golang.org/x/net/xsrftoken"
+)
+
+const (
+	// TokenKey is the form key used when sending the token as part of POST
+	// request
+	TokenKey = "xsrf-token"
+)
+
+// StorageService is an interface the framework users need to implement. This
+// will contain information about users of the web application, including their
+// IDs, needed in generating the XSRF token.
+type StorageService interface {
+	// TODO(@mihalimara22): add the parameters that the storage service needs in
+	// order to determine the user ID
+	GetUserID() (string, error)
+}
+
+// Plugin implements XSRF protection.
+//
+// TODO(@mihalimara22): Add Fetch Metadata support
+type Plugin struct {
+	//appKey uniquely identifies each registered server and should have high
+	//entropy as it will be used in the xsrf token generation algorithm.
```

Comment:
Move this comment to the godoc comment of `NewPlugin`. Here it won't be visible to the user.